app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationScreen.kt

```diff
@@ -168,21 +185,36 @@ fun ConversationScreen(
     )
 }
 
+@Suppress("LongParameterList")
 private fun startCallIfPossible(
     conversationCallViewModel: ConversationCallViewModel,
     showDialog: MutableState<ConversationScreenDialogType>,
     startCallAudioPermissionCheck: CallingAudioRequestFlow,
-    coroutineScope: CoroutineScope
+    coroutineScope: CoroutineScope,
+    conversationType: Conversation.Type,
+    onOpenOngoingCallScreen: () -> Unit
 ) {
     coroutineScope.launch {
         if (!conversationCallViewModel.hasStableConnectivity()) {
             showDialog.value = ConversationScreenDialogType.NO_CONNECTIVITY
         } else {
+            val canStartCall = (conversationType == Conversation.Type.ONE_ON_ONE ||
+                    (conversationType == Conversation.Type.GROUP && conversationCallViewModel.isConferenceCallingEnabled()))
```

Comment:
I would suggest to do this check in the `startCallUseCase` instead of here for two reasons:
1. We don't need to implement this again for different clients : iOS, web and JVM later
2. Checking if the user can make a call or not, is something internal, it's not an app usecase we don't even need to know about it in high level

Now `StartCallUseCase` is returning`Result` containing  `Success` and `SyncFailure`, we could add a new failure object there like `UnPaidPlanFailure` and handle the this failure there ?